Mage/src/main/java/mage/players/ManaPool.java

```diff
@@ -146,6 +152,7 @@ public boolean pay(ManaType manaType, Ability ability, Filter filter, Game game,
                 game.fireEvent(event);
                 usedManaToPay.increase(usableManaType);
```

Comment:
omg, there are another payment store: `usedManaToPay`

Usage of `usedManaToPay` must be changed too all around the code. Found that it was introduced for [Mycosynth Lattice](https://scryfall.com/card/bbd/241/mycosynth-lattice) and `may spend mana as though it were mana of any color`. So the real used mana must be find by `getUsedManaToPay()` not by `getPayment()`. There are few cards with `getPayment()` usage -- so it must be replaced by `getUsedManaToPay()` too.

Also `clearPaid()` miss from clear of `usedManaToPay`. Also methods with `payment` checks must be fixed to use `usedManaToPay` (example: `isColoredPaid`);

![shot_210119_070519](https://user-images.githubusercontent.com/8344157/104982749-b6600280-5a24-11eb-821c-4d077d8f7b73.png)

Reply:
alright, I think I need to write some tests for this to make sure it's all working right